flink/v1.17/flink/src/main/java/org/apache/iceberg/flink/FlinkCatalog.java

```diff
@@ -605,6 +659,56 @@ private static void commitChanges(
     transaction.commitTransaction();
   }
 
+  private static void commitChanges(
+      Table table,
+      String setLocation,
+      String setSnapshotId,
+      String pickSnapshotId,
+      List<TableChange> schemaChanges,
+      List<TableChange> propertyChanges) {
+    commitChanges(table, setSnapshotId, pickSnapshotId);
+
+    Transaction transaction = table.newTransaction();
+
+    if (setLocation != null) {
+      transaction.updateLocation().setLocation(setLocation).commit();
+    }
+
+    if (!schemaChanges.isEmpty()) {
+      UpdateSchema updateSchema = transaction.updateSchema();
+      FlinkAlterTableUtil.applySchemaChanges(updateSchema, schemaChanges);
+      updateSchema.commit();
```

Comment:
maybe `commit` action can be moved inside `FlinkAlterTableUtil`? then it would be more natural to move `applyManageSnapshots` into `FlinkAlterTableUtil` too as mentioned in a comment below.

Reply:
That's reasonable.